oximeter/db/src/client/mod.rs

```diff
@@ -1008,10 +1017,45 @@ impl Client {
 
     /// Given a list of timeseries by name, delete their schema and any
     /// associated data records from all tables.
+    ///
+    /// If the database isn't available or the request times out, this method
+    /// will continue to retry the operation until it succeeds.
     async fn expunge_timeseries_by_name(
         &self,
         replicated: bool,
         to_delete: &[TimeseriesName],
+    ) -> Result<(), Error> {
+        let op = || async {
+            self.expunge_timeseries_by_name_once(replicated, to_delete)
+                .await
+                .map_err(|err| match err {
+                    Error::DatabaseUnavailable(_) => {
+                        backoff::BackoffError::transient(err)
+                    }
+                    _ => backoff::BackoffError::permanent(err),
+                })
+        };
+        let notify = |error, delay| {
+            warn!(
+                self.log,
+                "failed to delete some timeseries";
+                "error" => ?error,
+                "retry_after" => ?delay,
+            );
+        };
+        backoff::retry_notify(
```

Comment:
is it, perhaps, worth using `retry_notify_ext` instead, so that we can also log the total retry count here?

Reply:
Thanks! Done in be18651bc